test/jdk/java/net/ProxySelector/LoopbackAddresses.java

```diff
@@ -63,15 +74,18 @@ public static void main(String[] args) {
             // to answer both for the loopback and "localhost".
             // Though "localhost" usually point to the loopback there is no
             // hard guarantee.
-            server = new TestHttpServer (new LoopbackAddresses(), 1, 10, 0);
-            ProxyServer pserver = new ProxyServer(InetAddress.getByName("localhost"), server.getLocalPort());
+            server = HttpServer.create(new InetSocketAddress(loopback, 0), 10);
+            server.createContext("/", new LoopbackAddresses());
```

Comment:
While here, we could use the new [HttpServer::create](https://download.java.net/java/early_access/jdk18/docs/api/jdk.httpserver/com/sun/net/httpserver/HttpServer.html#create(java.net.InetSocketAddress,int,java.lang.String,com.sun.net.httpserver.HttpHandler,com.sun.net.httpserver.Filter...)) overload that creates a server with a context (applies to all tests touched.) 

I also wonder if it would make sense to change the names of the classes that implement HttpHandler to "xyzHandler" for readability, and to create a separate class for the handler instead of using the test class.